app/src/main/java/com/m2049r/xmrwallet/service/WalletService.java

```diff
@@ -164,16 +165,20 @@ public void refreshed() { // this means it's synced
     private long lastDaemonStatusUpdate = 0;
     private long daemonHeight = 0;
     private Wallet.ConnectionStatus connectionStatus = Wallet.ConnectionStatus.ConnectionStatus_Disconnected;
-    private static final long STATUS_UPDATE_INTERVAL = 120000; // 120s (blocktime)
+    private static final long STATUS_UPDATE_INTERVAL_SYNCED = 120000; // 120s (blocktime)
+    private static final long STATUS_UPDATE_INTERVAL_SYNCING = 10000; // 10s
 
-    private void updateDaemonState(Wallet wallet, long height) {
+    private boolean updateDaemonState(Wallet wallet, long height, boolean receivedNewBlock) {
+        Wallet.ConnectionStatus startConnectionStatus = connectionStatus;
         long t = System.currentTimeMillis();
-        if (height > 0) { // if we get a height, we are connected
-            daemonHeight = height;
+        if (daemonHeight > 0 && height > 0 && (height > daemonHeight || receivedNewBlock)) {
+            if (height > daemonHeight)
+                daemonHeight = height;
             connectionStatus = Wallet.ConnectionStatus.ConnectionStatus_Connected;
             lastDaemonStatusUpdate = t;
         } else {
-            if (t - lastDaemonStatusUpdate > STATUS_UPDATE_INTERVAL) {
+            long statusUpdateInterval = wallet.isSynchronized() ? STATUS_UPDATE_INTERVAL_SYNCED : STATUS_UPDATE_INTERVAL_SYNCING;
+            if (daemonHeight == 0 || t - lastDaemonStatusUpdate > statusUpdateInterval) {
```

Comment:
Also calling out this part. Notice how I added `daemonHeight > 0` in the above if statement, and the first time `updateDaemonState` gets called, `daemonHeight == 0`, so it will fall into this if and make the request for the height. If the height [is already cached](https://github.com/monero-project/monero/blob/67e5ca9ad6f1c861ad315476a88f9d36c38a0abb/src/wallet/node_rpc_proxy.cpp#L148-L150), it won't be a problem.

The work over in https://github.com/monero-project/monero/pull/8076 *should* minimize trips to the daemon and ideally get the height cached so the client shouldn't need to make a trip to the daemon in this case. Looking into this as well.